### DISCUSSION TITLE
More interesting boundary conditions for the Swift-Hohenberg pattern formation model (biharmonic)

### DISCUSSION THREAD

**Discussion by aquillen** [kinnala/scikit-fem#1140] [Q&A]
I am interested in trying out different boundary conditions for a Swift-Hohenberg pattern formation model using Morley elements.  This PDE has a sum of Laplacian and biharmonic operators.  I am not sure how to set the different possible boundary conditions for Morley elements and would be very grateful for some examples or suggestions or pointers!  
The boundary identified with get_dofs() seems to have both field and normal derivatives, so perhaps they can be set separately (as if they were Dirichlet boundary conditions)?  The weak formulation has Laplacian and gradient of Laplacian terms on the boundary and I am not sure how I would set these. The simplest thing (meaning without any fussing with boundary) is in a notebook here: https://github.com/aquillen/Pattern_Formation_FEM/blob/main/fem_swift_hohenberg.ipynb

**Comment by kinnala** [comment]
Field values and normal derivatives can be set separately as you suggest as essential conditions by restricting the corresponding DOF values. You can do `basis.get_dofs().all('u')` to get the DOF indices matching just the field value or `basis.get_dofs().all('u^n')` to get normal derivative DOF indices.

Boundary conditions on the second and the third derivatives would need to be natural to the variational formulation used, similar to how conditions on the normal derivative are natural to the canonical variational formulation of the Poisson problem. So there isn’t any specific command to set those, you should correctly apply integration-by-parts and set correct terms to zero to enforce such conditions automatically.  This is similar to how it is done for the fourth-order Kirchhoff plate bending problem which also uses Morley elements.

**Reply by kinnala** [reply]
Let me add that (nonconforming) Morley might not be suitable in case you have explicit second or third order derivatives in the variational form since the basis is piecewise quadratic and already the Laplacian goes to zero elementwise. This could be circumvented by using a conforming basis such as Argyris element.

**Reply by aquillen** [reply]
This is very helpful! Thank you.  I will give these suggestions a try!
